sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/aio/_client_manager.py

```diff
@@ -0,0 +1,457 @@
+                )
+                return True, updated_sentinel
+        except HttpResponseError as e:
+            if e.status_code == 404:
+                if etag is not None:
+                    # If the sentinel is not found, it means the key/label was deleted, so we should refresh
+                    self.LOGGER.debug("Refresh all triggered by key: %s label %s.", key, label)
+                    return True, None
+            else:
+                raise e
+        return False, None
+
+    @distributed_trace
+    async def load_configuration_settings(
+        self, selects: List[SettingSelector], refresh_on: Dict[Tuple[str, str], str], **kwargs
+    ) -> Tuple[List[ConfigurationSetting], Dict[Tuple[str, str], str]]:
+        configuration_settings = []
+        sentinel_keys = kwargs.pop("sentinel_keys", refresh_on)
+        for select in selects:
+            configurations = self._client.list_configuration_settings(
+                key_filter=select.key_filter, label_filter=select.label_filter, **kwargs
+            )
+            async for config in configurations:
+                if isinstance(config, FeatureFlagConfigurationSetting):
+                    # Feature flags are ignored when loaded by Selects, as they are selected from
+                    # `feature_flag_selectors`
+                    pass
+                configuration_settings.append(config)
+                # Every time we run load_all, we should update the etag of our refresh sentinels
+                # so they stay up-to-date.
+                # Sentinel keys will have unprocessed key names, so we need to use the original key.
+                if (config.key, config.label) in refresh_on:
+                    sentinel_keys[(config.key, config.label)] = config.etag
+        return configuration_settings, sentinel_keys
+
+    @distributed_trace
+    async def load_feature_flags(
+        self, feature_flag_selectors: List[SettingSelector], feature_flag_refresh_enabled: bool, **kwargs
+    ) -> Tuple[List[FeatureFlagConfigurationSetting], Dict[Tuple[str, str], str], Dict[str, bool]]:
+        feature_flag_sentinel_keys = {}
+        loaded_feature_flags = []
+        # Needs to be removed unknown keyword argument for list_configuration_settings
+        kwargs.pop("sentinel_keys", None)
+        filters_used = {}
+        for select in feature_flag_selectors:
+            feature_flags = self._client.list_configuration_settings(
+                key_filter=FEATURE_FLAG_PREFIX + select.key_filter, label_filter=select.label_filter, **kwargs
+            )
+            async for feature_flag in feature_flags:
+                loaded_feature_flags.append(json.loads(feature_flag.value))
+                if not isinstance(feature_flag, FeatureFlagConfigurationSetting):
+                    # If the feature flag is not a FeatureFlagConfigurationSetting, it means it was selected by
+                    # mistake, so we should ignore it.
+                    continue
```

Comment:
If this is possible, should we be checking this in the sync version of `_client_manager`?

Reply:
So, I'm not sure why I need to add this. This throws an error in our type hinting if I don't check it, but it doesn't for sync. Any ideas.